msal/src/main/java/com/microsoft/identity/client/Prompt.java

```diff
@@ -41,20 +41,25 @@
      * <p>
      * toString override is to enable the correct protocol value of login to be returned instead of "force_login".
      */
-    FORCE_LOGIN,
+    LOGIN,
 
     /**
      * acquireToken will send prompt=consent to the authorize endpoint.  The user will be prompted to consent even if consent was granted before.
      */
-    CONSENT;
+    CONSENT,
+
+    /**
+     * acquireToken will not send the prompt parameter to the authorize endpoint.  The user may be prompted to login or to consent as required by the request.
+     */
+    WHEN_REQUIRED;
```

Comment:
`toString()` doesn't appear to handle this case

Reply:
same for `toOpenIdConnectPromptParameter()`

Reply:
Yes... because it's not meant to be converted to string.  I felt that throwing illegal argument exception would make that very clear.

Reply:
It has no corresponding OpenIdConnectPromptParameter value

Reply:
This should be an `UnsupportedOperationException` then, as there is no argument

Reply:
Also, not meant to be converted to a string sounds suspicious. What about logging?

Reply:
Let me see what I can do to achieve my goal in a different way.

Reply:
Fixed.